### ISSUE TITLE
Consider using the prefer-desctructuring in es6 plugin

### ISSUE THREAD

**Issue by pinussilvestrus** [bpmn-io/eslint-plugin-bpmn-io#1]
With the start of using [object destructuring in es6](http://exploringjs.com/es6/ch_destructuring.html) we used to use this possibility in most of our projects. Besides that, after an Hour of Code refactoring session it came clear we often mixed up certain ways of object destructuring  (cf. [snippet](https://github.com/bpmn-io/hour-of-code/blob/master/snippets/app_index.md)). To create a certain consistency we could consider using the [`prefer-destructuring`](https://eslint.org/docs/rules/prefer-destructuring) inside the es6 plugin.

**Comment by nikku** [comment]
We are operating in three modes, essentially:

* ES5: Just plain ES5
* ES6-kinda-ES5: Just plain ES5 + ES Modules so it compiles to ES 5 when piped through a modern module bundler
* ES6 (Camunda Modeler + dmn-js): FULL ES6 (no mercy style)

We cannot use destructuring in ES6-kinda-ES5 mode. We could however rename the modes or add new ones.

**Comment by pinussilvestrus** [comment]
After I gave it a try-out inside the [`camunda-modeler`](https://github.com/camunda/camunda-modeler/), I made the following conclusions:

Right now we have _79_ lint errors regarding the `prefer-destructuring` rule inside the application. I tried out the autofix option (which is quite new, cf. https://github.com/eslint/eslint/commit/5eb3121b82c1837da0c3021b7d9384bb30832e36). It seems that automatic fixes are not fit my expectations. For example following line

```js
const foo = options.foo;
const bar = options.bar;
```

will be refactored to this

```js
const { foo }  = options;
const { bar } = options;
```

The expected behavior would be (in my opinion):

```js
const {
   foo,
   bar
} = options;
```

On top, the autofix seems not to affect array-destructuring right now. The fact the autofix does not solve our problems, and we don't want to fix all occurrences manually, I would propose: Leave the code as it is, don't use this rule and refactor the occurrences in a lazy manner, which means every time we step over a missing destructuring, simply add it.
